pkg/asset/agent/installconfig.go

```diff
@@ -187,58 +188,53 @@ func warnUnusedConfig(installConfig *types.InstallConfig) {
 	switch installConfig.Platform.Name() {
 
 	case baremetal.Name:
+		defaultIc := &types.InstallConfig{Platform: types.Platform{BareMetal: &baremetal.Platform{}}}
```

Comment:
In the case where the provisioning network is disabled and the ProvisioningIP is set, the default ProvisioningNetworkCIDR is derived from network CIDRs in the install-config. We could avoid a warning in that case by starting with a copy of the InstallConfig, instead of a blank one. (However, there will already be warnings about ProvisioningNetwork=disabled and ProvisioningIP in that case, so this is not a big deal.)